spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -246,8 +246,19 @@ public SparkTable alterTable(Identifier ident, TableChange... changes) throws No
   public boolean dropTable(Identifier ident) {
     try {
       return isPathIdentifier(ident) ?
-          tables.dropTable(((PathIdentifier) ident).location()) :
-          icebergCatalog.dropTable(buildIdentifier(ident));
+          tables.dropTable(((PathIdentifier) ident).location(), false) :
+          icebergCatalog.dropTable(buildIdentifier(ident), false);
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
+      return false;
+    }
```

Comment:
Rather than maintaining this in 2 places, can you use a private `dropTableInternal` method that accepts a `boolean purge` argument? That will cut down on duplication.

Also, can you either fix 3.1 or 3.2 in this PR? It's easier to fix one version and port the changes once they are approved than to try to keep multiple versions in sync in a a single PR.

Reply:
done